contracts/core/space.cairo

```diff
@@ -521,3 +524,27 @@ func _catch_dust{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_p
     IDustContract.safeTransferFrom(dust_contract_address, contract_address, ship, token_id)
     return ()
 end
+
+@view
+func get_grid_state{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (grid_state_len: felt, grid_state: Cell*):
+    alloc_locals
+
+    let (local grid_state : Cell*) = alloc()
+    
+    let (grid_state_len) = _rec_fill_grid_state(0, grid_state, 0)
+    
+    return (grid_state_len, grid_state)
+end
+
+func _rec_fill_grid_state{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(grid_state_len: felt, grid_state: Cell*, index: felt) -> (len: felt):
```

Comment:
index and grid_state_len seem to have the same purpose

Reply:
Fixed